pkg/skaffold/jib/jib_init.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+// For testing
+var (
+	ValidateJibConfigFunc = ValidateJibConfig
+)
+
+const (
+	// JibGradle the name of the Jib Gradle Plugin
+	JibGradle = "Jib Gradle Plugin"
+	// JibMaven the name of the Jib Maven Plugin
+	JibMaven = "Jib Maven Plugin"
+)
+
+// Jib holds information about a Jib project
+type Jib struct {
+	BuilderName string `json:"-"`
+	Image       string `json:"image,omitempty"`
+	FilePath    string `json:"path,omitempty"`
+	Project     string `json:"project,omitempty"`
+}
+
+// Name returns the name of the builder
+func (j Jib) Name() string {
+	return j.BuilderName
+}
+
+// Describe returns the initBuilder's string representation, used when prompting the user to choose a builder.
+func (j Jib) Describe() string {
+	if j.Project != "" {
+		return fmt.Sprintf("%s (%s, %s)", j.BuilderName, j.Project, j.FilePath)
+	}
+	return fmt.Sprintf("%s (%s)", j.BuilderName, j.FilePath)
+}
+
+// CreateArtifact creates an Artifact to be included in the generated Build Config
+func (j Jib) CreateArtifact(manifestImage string) *latest.Artifact {
+	workspace := filepath.Dir(j.FilePath)
+
+	a := &latest.Artifact{ImageName: j.Image}
+	if j.Image == "" {
+		a.ImageName = manifestImage
```

Comment:
I forgot. So if a target image is configured in Jib (in `pom.xml` for example), then we can (or should) skip defining `image` in `skaffold.yaml`? This [repo example](https://github.com/GoogleContainerTools/skaffold/tree/master/examples) defines `image`s even if the images are configured in Jib.

Reply:
Never mind. I see `ImageName` is initialized at first.